src/components/settings/Settings.h

```diff
@@ -68,14 +73,34 @@ namespace Pinetime {
         return settings.screenTimeOut;
       };
 
-      void setWakeUpMode(WakeUpMode wakeUp) {
-        if (wakeUp != settings.wakeUpMode)
+      void resetWakeUpModes() {
+        settings.wakeUpMode.reset();
+      }
+
+      void setWakeUpMode(WakeUpMode wakeUp, bool enable) {
+        if (!isWakeUpModeOn(wakeUp) != enable) // Boolean XOR
```

Comment:
I think the ! before `isWakeUpModeOn` here needs to be removed as XOR is just `A != B`.

Reply:
good point. fixed in a sec.